### PR TITLE
Pass Opaque point to user

### PR DESCRIPTION
**Pull request by TApplencourt**
Rename common_data and name_2_dispatcher to `void *btx_handle`. 
The type of those object are just implementation detail and should not be exposed to user

### CODE REVIEW

**Comment by TApplencourt** [comment]
Sorry @DonAurelio  ugly PR. Please ignore the `display` and `example` modification. Those test are broken for now, they are not in the CI so I forgot to test it. 
We can fix them later :')

File: README.md [hunk 1]
```diff
@@ -4,9 +4,9 @@
 
 ## Filter & Sink
 
-Link with a object file who export the symbol `btx_usr_register_callbacks(btx_name_to_dispatcher_t**)`.
+Link with a object file who export the symbol `btx_usr_register_callbacks(void *)`.
 
-The implementation of `btx_usr_register_callbacks` should conssist of calls to `btx_register_callbacks_#{stream_class_name}_#{event_class_name}(name_to_dispatcher, &callbacks)`.
```
Comment (DonAurelio) [inline README.md:9]: 1. I think at some point you changed from `btx_usr_register_callbacks` to `btx_register_usr_callbacks`. 
2. Same `btx_register_callbacks_#{stream_class_name}_#{event_class_name}(name_to_dispatcher, &callbacks)` to `btx_register_callbacks_#{event_class_name}(name_to_dispatcher, &callbacks)`

Comment (TApplencourt) [reply README.md:9]: I think we need to improve the read a lot. I will correct this.

Comment (TApplencourt) [reply README.md:9]: Fixed

File: template/component.h.erb [hunk 2]
```diff
@@ -136,17 +136,16 @@ struct common_data_s {
 <% end %>
 
 /* Implemented by params.c */
-void btx_read_params(common_data_t *common_data, params_t *usr_params);
+void btx_read_params(void *btx_handle, params_t *usr_params);
```
Comment (DonAurelio) [inline template/component.h.erb:139]: Why sometimes `btx_handle_t` and others `void`?

Comment (TApplencourt) [reply template/component.h.erb:139]: Always `void *`, `btx_handle_t` is a mistake that I made in the example and display. Has to tired to fix, but will do

File: example/SINK.xprof/my_callbacks.cpp [hunk 3]
```diff
@@ -39,7 +39,7 @@ static void bernard(struct common_data_s *common_data, void *usr_data, int32_t i
     data->count += 1;
 }
 
-void btx_register_usr_callbacks(name_to_dispatcher_t** name_to_dispatcher) {
-  btx_register_callbacks_lttng_ust_ze_profiling_event_profiling_results(name_to_dispatcher, &roger);
-  btx_register_callbacks_lttng_ust_ze_profiling_event_profiling_results(name_to_dispatcher, &bernard);
+void btx_register_usr_callbacks(btx_handle_t** btx_handle) {
```
Comment (DonAurelio) [inline example/SINK.xprof/my_callbacks.cpp:42]: I see, you are using the same replacement "btx_handle" for `common_data` and `name_to_dispatch`. I understand your complain, but here we are mixind two things that even the user would not know and does not care, however, if somebody try to undestand why two different things end up with the same type_name, is that conceptually right?.

File: template/dispatch.c.erb [hunk 4]
```diff
@@ -17,16 +18,17 @@ btx_dispatch_<%= e.name_sanitized %>(UT_array *callbacks,
   // Their type are declared in 'dispatch.h'
   <%= e.name_sanitized %>_callback_f **p = NULL;
   while ((p = utarray_next(callbacks, p))) {
-    (*p)(common_data, common_data->usr_data,
+    (*p)((void *)common_data, common_data->usr_data,
```
Comment (DonAurelio) [inline template/dispatch.c.erb:21]: I aggree the idea of put more meaningfull names, but if that may add more complexity to the code, I am not sure if it worth it, even when we have to cast everywhere, what do you think? or maybe I am not seeing the big picture.

File: example/SINK.xprof/my_callbacks.cpp (hunk 3 above)
Comment (TApplencourt) [reply example/SINK.xprof/my_callbacks.cpp]: It's an implementation detail. The user should not care. 
Maybe one day we will but `name_to_dispatcher` into `common_data`. Better to just hide it.

File: template/dispatch.c.erb (hunk 4 above)
Comment (TApplencourt) [reply template/dispatch.c.erb:21]: It's more about "separation of concern." The user of metababel, should not need to care about the internal that we use inside metababel.

File: template/component.h.erb (hunk 2 above)
Comment (DonAurelio) [reply template/component.h.erb:139]: I can do it, to familiarize with the new naming.

**Pull request merged**